app/src/main/java/ch/epfl/sdp/blindly/recording/AudioLibraryAdapter.kt

```diff
@@ -0,0 +1,218 @@
+        return ViewHolder(view)
+    }
+
+    // Replace the contents of a view (invoked by the layout manager)
+    @RequiresApi(Build.VERSION_CODES.N)
+    override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
+        // Get element from your dataset at this position and replace the
+        // contents of the view with that element
+        viewHolder.recordName.text = recordList[position].name
+        viewHolder.recordDuration.text = recordList[position].durationText
+        viewHolder.nameDurationLayout.setOnClickListener {
+            val show: Boolean = toggleLayout(!recordList[position].isExpanded, it, viewHolder.expandableLayout)
+            recordList[position].isExpanded = show
+        }
+
+        val playBar = viewHolder.playBar
+        val playTimer = viewHolder.playTimer
+        val remainingTimer = viewHolder.remainingTimer
+        val playPauseButton = viewHolder.playPauseButton
+        val movePlayBarThread = createPlayBarThread(playBar)
+
+        setCountDownTimer(remainingTimer)
+
+        viewHolder.playPauseButton.setOnClickListener {
+            if (isPlayerStopped) {
+                createMediaPlayer(recordList[position].filePath)
+                remainingTimer.base = SystemClock.elapsedRealtime() + mediaPlayer!!.duration.toLong()
+
+                mediaPlayer?.setOnCompletionListener {
+                    mediaPlayer?.stop()
+
+                    setCompletedAudioView(playTimer, remainingTimer, playPauseButton)
+                }
+            }
+            if (!mediaPlayer!!.isPlaying) {
+                mediaPlayer?.start()
+
+                if (!isPlayerPaused) {
+                    playTimer.base = SystemClock.elapsedRealtime()
+                }
+                setPlayView(playTimer, remainingTimer, playBar, movePlayBarThread, playPauseButton)
+            } else {
+                mediaPlayer?.pause()
+
+                setPauseView(playTimer, remainingTimer, playPauseButton)
+            }
+        }
+
+        viewHolder.selectButton.setOnClickListener {
+            mediaPlayer?.release()
+            saveRecording(position)
+            startProfileFinished()
+        }
+    }
+
+    private fun saveRecording(position: Int) {
+        val filePath = recordList[position].filePath
+        val newName = "PresentationAudio.3gp"
+        val currentRecording = File(filePath)
+        currentRecording.copyTo(File("${context.filesDir.absolutePath}/$newName"), overwrite = true)
+    }
+
+    private fun createMediaPlayer(filePath: String) {
+        mediaPlayer = MediaPlayer().apply {
+            setDataSource(filePath)
+            prepare()
+        }
+    }
+
+    private fun createPlayBarThread(playBar: SeekBar): Runnable {
+        return object : Runnable {
+            override fun run() {
+                if (mediaPlayer?.isPlaying == true) {
+                    val newMediaPos = mediaPlayer!!.currentPosition
+                    val newMediaMax = mediaPlayer!!.duration
```

Comment:
I think the val here aren't necessary, it would be more logic to use the values directly instead of storing them